app/waiting-approval/page.tsx

```
@@ -0,0 +1,58 @@
+        router.push("/");
+      }
+    }, [user, isApproved, router]);
+
+  const handleLogout = async () => {
+    await supabase.auth.signOut();
+    router.push("/login");
+  };
+
+  // let the effect below handle navigation instead of pushing in render
+  if (!user) return null;
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)